File: lib/biz/spec-iot/src/main/java/site/addzero/biz/spec/iot/protocol/s7/S7DataArea.java
package site.addzero.biz.spec.iot.protocol.s7;

/**
 * Common S7 data areas.
 */
public enum S7DataArea {

    DB("DB"),
    I("I"),
    Q("Q"),
    M("M"),
    V("V");

    private final String code;

    S7DataArea(String code) {
        this.code = code;
    }

    public String getCode() {
        return code;
    }
}

File: lib/biz/spec-iot/src/main/kotlin/site/addzero/biz/spec/iot/protocol/s7/S7DataArea.kt
package site.addzero.biz.spec.iot.protocol.s7

/**
 * Common S7 data areas.
 */
enum class S7DataArea(
    val code: String,
) {
    DB("DB"),
    I("I"),
    Q("Q"),
    M("M"),
    V("V"),
}
